lib/spack/spack/test/cmd/find.py

```diff
@@ -331,3 +331,10 @@ def test_find_loaded(database, working_env):
     output = find('--loaded')
     expected = find()
     assert output == expected
+
+
+@pytest.mark.regression('10019')
+def test_find_packages_with_dependencies(database):
```

Comment:
I appreciate the test but since the bug was with `Database` IMO the test should be more direct (i.e. execute and test the result of a `Database` function vs. executing and testing the output of a command).

Reply:
Fair. In the process of moving the test I also discovered that the error is triggered only by conditional dependencies, so added a mock package that models that.